src/main/scala/scorex/core/network/DeliveryTracker.scala

```diff
@@ -82,22 +86,25 @@ class DeliveryTracker(system: ActorSystem,
     tryWithLogging {
       val checks = requested(modifierId).checks + 1
       setUnknown(modifierId)
-      if (checks < maxDeliveryChecks) setRequested(modifierId, modifierTypeId,  Some(cp), checks)
+      if (checks < maxDeliveryChecks) setRequested(modifierId, modifierTypeId, cp, checks)
       else throw new StopExpectingError(modifierId, checks)
     }
 
   /**
     * Set status of modifier with id `id` to `Requested`
     */
-  def setRequested(id: ModifierId, typeId: ModifierTypeId, supplierOpt: Option[ConnectedPeer], checksDone: Int = 0)
-                  (implicit ec: ExecutionContext): Unit =
+  private def setRequested(id: ModifierId, typeId: ModifierTypeId, supplier: ConnectedPeer, checksDone: Int = 0)
+                          (implicit ec: ExecutionContext): Unit =
     tryWithLogging {
       require(isCorrectTransition(status(id), Requested), s"Illegal status transition: ${status(id)} -> Requested")
-      val cancellable = system.scheduler.scheduleOnce(deliveryTimeout, nvsRef, CheckDelivery(supplierOpt, typeId, id))
-      requested.put(id, RequestedInfo(supplierOpt, cancellable, checksDone))
+      val cancellable = system.scheduler.scheduleOnce(deliveryTimeout, nvsRef, CheckDelivery(supplier, typeId, id))
+      requested.put(id, RequestedInfo(supplier, cancellable, checksDone)) match {
+        case Some(_) => //we already had this modifier, it is counted
```

Comment:
It seems to be a case when we increment the `checksDone`.

But what if we have re-requested the `id` from different peer?
From the current class interface perspective it seems to be possible. So in this case, we will never `decrementPeerLimitCounter` for the old peer.
Let's add this check.